packages/airnode-protocol/contracts/WIP/Airnode.sol

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/Multicall.sol";
+import "./WithdrawalUtils.sol";
+import "./interfaces/IAirnode.sol";
+
+contract Airnode is Multicall, WithdrawalUtils, IAirnode {
+    using ECDSA for bytes32;
+
+    struct Template {
+        address airnode;
+        bytes32 endpointId;
+        bytes parameters;
+    }
+
+    mapping(bytes32 => Template) public templates;
+
+    mapping(address => mapping(address => bool))
+        public sponsorToRequesterToSponsorshipStatus;
+
+    mapping(address => uint256) public requesterToRequestCountPlusOne;
+
+    mapping(bytes32 => bytes32) private requestIdToFulfillmentParameters;
+
+    function createTemplate(
+        address airnode,
+        bytes32 endpointId,
+        bytes calldata parameters
+    ) external returns (bytes32 templateId) {
+        templateId = keccak256(
```

Comment:
No longer reverting if `airnode` is `address(0)` to be consistent with `makeRequest()`